src/miner.cpp

```diff
@@ -573,9 +551,14 @@ void GenerateBitcoins(bool fGenerate, CWallet* pwallet, int nThreads)
     if (nThreads == 0 || !fGenerate)
         return;
 
+    CScript coinbaseScript;
+    GetMainSignals().ScriptForMining(coinbaseScript);
```

Comment:
I think we can very reasonably assume that boost::signals2 will not fundamentally change their design and API in a way that will only remain compatible for void outputs.